velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -485,6 +485,119 @@ TEST_F(DateTimeFunctionsTest, yearTimestampWithTimezone) {
           "year(c0)", std::nullopt, std::nullopt));
 }
 
+TEST_F(DateTimeFunctionsTest, weekDate) {
+  const auto week = [&](std::optional<Date> date) {
+    return evaluateOnce<int64_t>("week(c0)", date);
+  };
+  auto parseDate = [&](const StringView& dateStr) {
+    Date returnDate;
+    parseTo(dateStr, returnDate);
+    return returnDate;
+  };
+
+  EXPECT_EQ(1, week(parseDate("1919-12-31")));
+  EXPECT_EQ(1, week(parseDate("1920-01-01")));
+  EXPECT_EQ(1, week(parseDate("1920-01-04")));
+  EXPECT_EQ(2, week(parseDate("1920-01-05")));
+  EXPECT_EQ(53, week(parseDate("1960-01-01")));
+  EXPECT_EQ(53, week(parseDate("1960-01-03")));
+  EXPECT_EQ(1, week(parseDate("1960-01-04")));
+  EXPECT_EQ(1, week(parseDate("1969-12-31")));
+  EXPECT_EQ(1, week(parseDate("1970-01-01")));
+  EXPECT_EQ(1, week(parseDate("0001-01-01")));
+  EXPECT_EQ(52, week(parseDate("9999-12-31")));
+}
+
+TEST_F(DateTimeFunctionsTest, week) {
+  const auto week = [&](const StringView& time) {
+    auto timestampInSeconds = util::fromTimeString(time) / 1'000'000;
+    return evaluateOnce<int64_t>(
+        "week(c0)",
+        std::make_optional(Timestamp(timestampInSeconds * 100'000'000, 0)));
+  };
+
+  EXPECT_EQ(1, week("00:00:00"));
+  EXPECT_EQ(10, week("11:59:59"));
+  EXPECT_EQ(51, week("06:01:01"));
+  EXPECT_EQ(24, week("06:59:59"));
+  EXPECT_EQ(27, week("12:00:01"));
+  EXPECT_EQ(7, week("12:59:59"));
+}
+
+TEST_F(DateTimeFunctionsTest, weekTimestampWithTimezone) {
+  const auto week = [&](const StringView& time, const StringView& timezone) {
+    auto timestampInSeconds = util::fromTimeString(time) / 1'000'000;
+    return evaluateWithTimestampWithTimezone<int64_t>(
+        "week(c0)", timestampInSeconds * 100'000'000, timezone);
+  };
+
+  EXPECT_EQ(1, week("00:00:00", "-12:00"));
+  EXPECT_EQ(1, week("00:00:00", "+12:00"));
+  EXPECT_EQ(47, week("11:59:59", "-12:00"));
+  EXPECT_EQ(47, week("11:59:59", "+12:00"));
+  EXPECT_EQ(33, week("06:01:01", "-12:00"));
+  EXPECT_EQ(34, week("06:01:01", "+12:00"));
+  EXPECT_EQ(47, week("12:00:01", "-12:00"));
+  EXPECT_EQ(47, week("12:00:01", "+12:00"));
+}
+
+TEST_F(DateTimeFunctionsTest, weekOfYearDate) {
```

Comment:
Since week_of_year is an alias of the "week" consider dropping these test cases or refactor to avoid copy-paste.

Reply:
Thanks for the suggestion, used bitwise AND between the results of ```week``` and ```week_of_year``` in the week lambdas to ensure both return the same expected value.